### PR TITLE
Issue 719: register spotfleet instances with ALB target group(s)

### PR DESCRIPTION
**Pull request by Cptn727**

### CODE REVIEW

File: core/controlplane/config/templates/cloud-config-worker [hunk 1]
```diff
@@ -783,6 +799,38 @@ write_files:
...
+
+      sudo rkt run \
+        --volume=ssl,kind=host,source=/etc/kubernetes/ssl,readOnly=false \
+        --mount=volume=ssl,target=/etc/kubernetes/ssl \
+        --uuid-file-save=/var/run/coreos/add-to-target-groups.uuid \
+        --volume=dns,kind=host,source=/etc/resolv.conf,readOnly=true --mount volume=dns,target=/etc/resolv.conf \
+        --net=host \
+        --trust-keys-from-https \
+        --insecure-options=ondisk \
+        {{.AWSCliImage.Options}}{{.AWSCliImage.RktRepo}} --exec=/bin/bash -- \
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-worker]: can you do `while ! rkt run ... ` as we do elsewhere to cater for intermittent image pull and API calls failures

File: core/controlplane/config/templates/cloud-config-worker [hunk 2]
```diff
@@ -783,6 +799,38 @@ write_files:
...
+{{if .Experimental.TargetGroup.Enabled}}
+  - path: /opt/bin/add-to-target-groups
+    owner: root:root
+    permissions: 0700
+    content: |
+      #!/bin/bash -e
+
+      instance_id=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+
+      sudo rkt run \
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-worker]: no need for sudo here

File: core/controlplane/config/templates/cloud-config-worker [hunk 3]
```diff
@@ -783,6 +799,38 @@ write_files:
...
+        --volume=ssl,kind=host,source=/etc/kubernetes/ssl,readOnly=false \
+        --mount=volume=ssl,target=/etc/kubernetes/ssl \
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-worker]: no need for `/etc/kubernetes/ssl` mounts

File: core/controlplane/config/templates/cloud-config-worker [hunk 4]
```diff
@@ -783,6 +799,38 @@ write_files:
...
+        --uuid-file-save=/var/run/coreos/add-to-target-groups.uuid \
+        --volume=dns,kind=host,source=/etc/resolv.conf,readOnly=true --mount volume=dns,target=/etc/resolv.conf \
+        --net=host \
+        --trust-keys-from-https \
+        --insecure-options=ondisk \
+        {{.AWSCliImage.Options}}{{.AWSCliImage.RktRepo}} --exec=/bin/bash -- \
+          -vxec \
+          'echo adding this spot instance to target groups
+           instance_id="'$instance_id'"
+           tgs=({{range $i, $tg := .Experimental.TargetGroup.Arns}}"{{$tg}}" {{end}})
```
Comment (redbaron) [inline core/controlplane/config/templates/cloud-config-worker]: you don't use `$i` here so can use single var range loop

**Comment by mumoshu** [comment]
@Cptn727 Hi, thank you very much for your contribution 👍 
I seems like you need to sign CLA.
Also, could you addresses comments made by @redbaron?
Anyways, I guess I can adapt this PR and make necessary changes on my side if you'd want me to do so.

**Comment by mumoshu** [comment]
I'm taking over your work in #821.
Thank you very much for your contribution @Cptn727 👍

**Pull request closed without merging**